### DISCUSSION TITLE
SMB PS2 server through powerline

### DISCUSSION THREAD

**Discussion by vincenzolegrottaglie** [retronas/discussions#128] [Q&A]
Do you think that is possible to run smb server from rpi and connect it to the ps2 ethernet port through a powerline?

**Comment by sairuk** [comment]
as in EoP adapters? sure, performance will depend on the quality of the connection which will be influenced by the quality of the lines etc.

**Reply by vincenzolegrottaglie** [reply]
So I will connect my rpi to router using Wi-Fi, then connecting the rpi ethernet to the adapter, so the ps2 to the other adapter?

**Reply by sairuk** [reply]
sounds like what i'd expect to be done if using EoP adapters, try it an see how you go

**Reply by vincenzolegrottaglie** [reply]
I'll try and give you a feedback

**Reply by vincenzolegrottaglie** [reply]
![IMG_5456](https://github.com/user-attachments/assets/4bf90f44-0d9d-44db-85ff-cf31378a8551)
I'm having issue to connect to the server. I'm trying the direct mode (RPi to PS2 ethernet cable). It says "300: Cannot connect to SMB server.". The password of SMB server is the same of the user one (Retronas set the same password, right?) and I can log using MacOS or Windows to the \\retrosmb folder or 192.168.1.23 like shown in the photo with that credentials. Can you help me?

**Reply by sairuk** [reply]
that looks fine in theory but check the ip address of the ethernet device on retronas, if the ps2 is directly connected to the retronas over ethernet then you'll need to confirm the ethernet ip is `.23` and not something else.

run `ip a` and look for `eth0` or `enp0s25` or the like then check the `inet` address

You don't mention if you are connecting windows and mac over wifi or not, in the default setup samba should be listening on all addresses.

**Reply by vincenzolegrottaglie** [reply]
The rpi ip address is correct. I can log on cockpit using that address. Mac is over wifi, windows is over ethernet. Checking smb service it says is active. Does the ps2 obtain the ip from rpi? Maybe is that one wrong?

**Reply by sairuk** [reply]
you have the address set to static in your screenshot so no it doesn't obtain the address from the pi, nor do we support DHCP by default on retronas. 
* each of the networking interfaces on the rpi will have separate addresses, Confirm which is which and make sure the ps2 smb address is correct
* check you can you access the ps2 share directly from windows/mac using the same cable/ip/share/user/pass as the ps2 is configured to use
* `ping 192.168.1.250` from retronas terminal, see if you get a response, if not your network/configuration is broken.

So far as retronas functionality is concerned if windows/mac is OK retronas samba is working as expected.

**Reply by vincenzolegrottaglie** [reply]
![Screenshot 2024-10-15 alle 00 07 38](https://github.com/user-attachments/assets/aa8d4b85-aa19-4070-aae8-eb5ca9c63705)
Here the problem. Host is not reachable. I set that IP randomly. Is there some logic to assign it? I'll try another ethernet cable meanwhile

**Reply by sairuk** [reply]
as i mentioned earlier, you can now see in that screenshot that there is no inet address on `eth0` so there wont be a route to that ip .... you'll need to set a static ip on eth0 for communication over rpi ethernet. assuming the ps2 is connected directly to the ethernet port on the rpi

configure a static ip address on `eth0 ` this discussion probably good enough to get you setup https://forums.raspberrypi.com/viewtopic.php?t=366025

I would suggest you put your devices eth0/ps2 in their own subnet tho unless you expect your main network to be able to talk directly to the ps2, if so you might as well plug the cable into whatever the windows machine is plugged into so it can all work properly

retronas has nothing to do with rpi networking by default, you are expected to provide your own networked rpi atm, that said there are new experimental features for managing networking in testing atm.

**Reply by vincenzolegrottaglie** [reply]
The ip is now set up: inet 192.168.2.1/24 brd 192.168.2.255 scope global eth0 (changed also on OPL network settings under ps2 ip address)

What about gateway and dns server? Which IP should I set? I still receive error 300: Cannot connect to SMB server

**Reply by sairuk** [reply]
use 192.168.2.1 although there is no actual DNS service installed on retronas by default either, as per previous comments around networking. noting direct mode will not get you internet access automagically. Set the ps2 to an IP address in the 192.168.2.x net.

after you've set the ip correctly on retronas you will need to restart samba service on the retronas unit as well, this can be completed through the menus or by rebooting the unit

**Reply by vincenzolegrottaglie** [reply]
Sorry for the late reply but I was away for work. I still cannot reach the SMB server on PS2.
This is were I set the IP for eth0: `sudo nano /etc/network/interfaces.d/eth0`.
![Screenshot 2024-10-18 alle 19 40 26](https://github.com/user-attachments/assets/43873490-92cc-477c-89b0-e1084074b69c)

This is what I get using `ip addr` after reboot:
![Screenshot 2024-10-18 alle 19 40 07](https://github.com/user-attachments/assets/bb8e2ff2-af6d-4bcf-922a-6394fcab3dad)

This is what I have on OPL:
![IMG_5527](https://github.com/user-attachments/assets/94a0f5f7-5510-42d8-8794-0ec957c9306c)

OPL version is 1.2.0 beta. Could this cause a problem? I had another server and worked fine. So I don't think that this particular version can harm to the connection.

From terminal I can ping 192.168.2.2 without issues.

Samba server works as expected too:
![Screenshot 2024-10-18 alle 19 50 40](https://github.com/user-attachments/assets/b5d3c7e2-da4a-4233-899f-8dbe6e2d4c12)

Thank you for your patience

**Comment by sairuk** [comment]
your ps2 and eth0 on the pi cannot have the same ip address and be expected to work.
